c/meterpreter/workspace/metsrv/CMakeLists.txt

```diff
@@ -0,0 +1,144 @@
+cmake_minimum_required(VERSION 3.0.0 FATAL_ERROR)
+
+################### Variables. ####################
+# Change if you want modify path or other values. #
+###################################################
+
+set(PROJECT_NAME metsrv)
+# Output Variables
+set(OUTPUT_DEBUG ../../output/Debug/)
+set(OUTPUT_RELEASE ../../../output/Release/)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${OUTPUT_RELEASE})
+
+# Folders files
+set(CPP_DIR_1 ../../source/server/win)
+set(CPP_DIR_2 ../../source/server)
+set(HEADER_DIR_1 ../../source/server)
+set(HEADER_DIR_2 ../../source/server/win)
+set(HEADER_DIR_3 ../../source/common/)
+set(HEADER_DIR_4 ../../source/ReflectiveDLLInjection/common)
+
+############## CMake Project ################
+#        The main options of project        #
+#############################################
+
+project(${PROJECT_NAME} C)
+
+# Define Release by default.
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "Release")
+  message(STATUS "Build type not specified: Use Release by default.")
+endif(NOT CMAKE_BUILD_TYPE)
+
+# Definition of Macros
+add_definitions(
+   -D__GNUC__
+   -DNDEBUG 
+   -DUNICODE
+   -D_UNICODE
+)
+
+############## Artefacts Output #################
+# Defines outputs , depending Debug or Release. #
+#################################################
+
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+  set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${OUTPUT_DEBUG}")
+  set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${OUTPUT_DEBUG}")
+  set(CMAKE_EXECUTABLE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${OUTPUT_DEBUG}")
+else()
+  set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${OUTPUT_RELEASE}")
+  set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${OUTPUT_RELEASE}")
+  set(CMAKE_EXECUTABLE_OUTPUT_DIRECTORY "${OUTPUT_RELEASE}")
+endif()
+
+################### Dependencies ##################
+# Add Dependencies to project.                    #
+###################################################
+
+option(BUILD_DEPENDS 
+   "Build other CMake project." 
+   OFF 
+)
+
+# Dependencies : disable BUILD_DEPENDS to link with lib already build.
+if(BUILD_DEPENDS)
+   add_subdirectory(platform/cmake/backcompat ${CMAKE_BINARY_DIR}/backcompat)
+   add_subdirectory(platform/cmake/common ${CMAKE_BINARY_DIR}/common)
+   add_subdirectory(platform/cmake/ReflectiveDLLInjection ${CMAKE_BINARY_DIR}/ReflectiveDLLInjection)
+else()
+   link_directories(${OUTPUT_RELEASE})
+   link_directories(../../output/Release)
+   link_directories(/usr/x86_64-w64-mingw32/lib/)
+endif()
+
+################# Flags ################
+# Defines Flags for Windows and Linux. #
+########################################
+SET(CMAKE_SYSTEM_NAME Windows)
```

Comment:
I think we should pull all of the compiler specific stuff out of the `CMakeList.txt` file and put it in a toolchain file. We can then have a different toolchain file for x64 and x86. e.g: https://github.com/berenm/cmake-extra/blob/master/toolchain/x86_64-windows-mingw-clang.cmake